rxlifecycle/src/main/java/com/trello/rxlifecycle/TakeUntilGenerator.java

```diff
@@ -21,19 +22,37 @@ public Boolean call(T lifecycleEvent) {
 
     @NonNull
     @CheckResult
-    static <T> Observable<Boolean> takeUntilCorrespondingEvent(@NonNull final Observable<T> lifecycle,
-                                                               @NonNull final Func1<T, T> correspondingEvents) {
+    static <T> Observable<Boolean> takeUntilCorrespondingEvent(
+            @NonNull final Observable<T> lifecycle,
+            @NonNull final Func1<T, T> correspondingEvents) {
         return Observable.combineLatest(
-            lifecycle.take(1).map(correspondingEvents),
-            lifecycle.skip(1),
-            new Func2<T, T, Boolean>() {
-                @Override
-                public Boolean call(T bindUntilEvent, T lifecycleEvent) {
-                    return lifecycleEvent.equals(bindUntilEvent);
+                lifecycle
+                        .take(1)
+                        .map(correspondingEvents)
+                        .ambWith(Observable.<T>just(null))
```

Comment:
I don't want `BehaviorSubjects` being passed around. If only there were a way to enforce the idea of having to use a `replay(1)` stream. Hmm.